### PR TITLE
Map commands & custom hud static menu

### PR DESCRIPTION
**Pull request by braem**
Closes https://github.com/momentum-mod/game/issues/1032
Closes https://github.com/momentum-mod/game/issues/1031

Decided to throw up a PR before reviewing that issue as this is a small change.

- Moved menus into their own folder, `cfg/menus/`
- Moved all hud static menus into a single file, `hud_menus.cpp`
- Added a menu for map commands, useful for mappers. Includes these commands by default:
  - `toggle mat_wireframe 3 0`
  - `showtriggers_toggle`
  - `toggle mat_fullbright`
  - `toggle r_drawclipbrushes 2 0`
  - `toggle mat_drawwater`
  - `toggle snd_show`
  - `toggle r_drawentities`
  - `mat_reloadallmaterials`
  - `toggle sv_cheats`
- Added a menu intended for custom commands
- Renamed `showRuler` to `mom_ruler_show` (to match saveloc menu naming scheme)

### Checklist
- [x] **I have thoroughly tested all of the code I have modified/added/removed to ensure something else did not break**
- [x] If there is a localization token change, I have updated the `momentum_english_ref.res` file with the changes, ran `tokenizer.py` to generate an up-to-date localization file, and have committed both the `.res` file changes and the new localization `.txt` file
- [x] If I introduced new h/cpp files, I have added them to the appropriate project's VPC file (`server_momentum.vpc` / `client_momentum.vpc` / etc)
- [x] If I have added or modified any visual assets (models, materials, panels, effects, etc), I have taken screenshots / videos of them and attached them to this PR directly (screenshots uploaded through github, videos uploaded to youtube and linked)
- [x] If I have modified any console command, console variable, or momentum entity, I have opened an issue (or a PR) for it in the [Momentum Mod documentation repository](https://github.com/momentum-mod/docs)
- [x] My commits are relatively small and scoped to the best of my ability
- [x] My branch has a clear history of changes that can be easy to follow when being reviewed commit-by-commit
- [x] My branch is functionally complete; the only changes to be done will be those potentially requested in code review

### CODE REVIEW

**Comment by braem** [comment]
Implemented a system for `hud_menu_static`'s. This takes the filenames in `cfg/menus/` for naming. 

Includes commands:
- `mom_hud_menu_show`, which autocompletes based on filenames. Strips the extension and "_default" off
- `mom_hud_menu_hide`
- `mom_hud_menu_reload`, which reloads the filenames for autocomplete.

Piggybacks off of `hud_menu_static` in terms of executing a command when the menu is closed. Figured this is best as all the file parsing is done there.

File: mp/src/game/client/momentum/ui/menus/hud_menu_static_system.cpp [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+    return pMenu->IsMenuDisplayed(pszMenuName);
+}
+
+void CHudMenuStaticSystem::ReloadMenuList()
+{
+    m_vecMenuNames.RemoveAll();
+
+    FileFindHandle_t findHandle;
+    const char *pFilename = g_pFullFileSystem->FindFirstEx("cfg/menus/*.*", "MOD", &findHandle);
```
Comment (Gocnak) [inline mp/src/game/client/momentum/ui/menus/hud_menu_static_system.cpp:114]: They're all going to be res files so you'll want to instead do

```suggestion
    const char *pFilename = g_pFullFileSystem->FindFirstEx("cfg/menus/*.res", "MOD", &findHandle);
```

File: mp/src/game/client/momentum/ui/menus/hud_menu_static_system.h [hunk 2]
```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+class CHudMenuStaticSystem : public CAutoGameSystem
+{
+  public:
+    CHudMenuStaticSystem();
+
+    static void ShowMenu(const char *pszFileName);
+    static void HideMenu();
+    static bool IsMenuDisplayed(const char *pszMenuName);
```
Comment (Gocnak) [inline mp/src/game/client/momentum/ui/menus/hud_menu_static_system.h:10]: These don't need to be static, you have a global accessor with `g_pHudMenuStaticSystem`

File: mp/src/game/client/momentum/ui/menus/hud_menu_static_system.cpp [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+}
+
+CON_COMMAND(mom_hud_menu_hide, "Hides the currently open hud menu if there is one.")
+{
+    g_pHudMenuStaticSystem->HideMenu();
+}
+
+static int HudMenuCompletion(const char *pPartial, char commands[COMMAND_COMPLETION_MAXITEMS][COMMAND_COMPLETION_ITEM_LENGTH]);
+
+CON_COMMAND_F_COMPLETION(mom_hud_menu_show, "Opens a hud menu. Accepts the hud menu itself by file name. See cfg/menus/", 0, HudMenuCompletion)
```
Comment (Gocnak) [inline mp/src/game/client/momentum/ui/menus/hud_menu_static_system.cpp:21]: You can put this below the static func and join the declaration and definition.

File: mp/src/game/client/momentum/ui/menus/hud_menu_static.cpp [hunk 4]
```diff
@@ -32,6 +32,12 @@ bool CHudMenuStatic::ShouldDraw() { return CHudElement::ShouldDraw() && m_bMenuD
 
 bool CHudMenuStatic::IsMenuDisplayed() { return m_bMenuDisplayed && m_bMenuTakesInput; }
 
+bool CHudMenuStatic::IsMenuDisplayed(const char *pszName)
+{
+    return IsMenuDisplayed() && FStrEq(pszName, m_pszFilename) ||
+           FStrEq(pszName, m_kvFromFile->GetString("menu_name", ""));
```
Comment (Gocnak) [inline mp/src/game/client/momentum/ui/menus/hud_menu_static.cpp:38]: Why not just use `m_kvFromFile->GetName()`? It'd save ya the effort of duplicating the menu name.

File: mp/src/game/client/momentum/ui/menus/hud_menu_static.h [hunk 5]
```diff
@@ -47,6 +48,8 @@ class CHudMenuStatic : public CHudElement, public vgui::Panel
   private:
     void (*SelectFunc)(int);
     void (*CloseFunc)();
```
Comment (Gocnak) [inline mp/src/game/client/momentum/ui/menus/hud_menu_static.h:50]: Is there any reason to be a close func if the subclasses were deleted?

**Review by Gocnak — CHANGES_REQUESTED** [review]

File: mp/src/game/client/momentum/ui/menus/hud_menu_static.h (hunk 5 above)
Comment (braem) [reply mp/src/game/client/momentum/ui/menus/hud_menu_static.h:50]: There isn't no, unless something else wants to use it in the future. Safe to remove?

Comment (Gocnak) [reply mp/src/game/client/momentum/ui/menus/hud_menu_static.h:50]: Yep should be safe to remove.

File: mp/src/game/client/momentum/ui/menus/hud_menu_static_system.cpp (hunk 1 above)
Comment (braem) [reply mp/src/game/client/momentum/ui/menus/hud_menu_static_system.cpp]: I assume you mean vdf files. I changed it to explicitly reference them.

File: mp/src/game/client/momentum/ui/menus/hud_menu_static.cpp (hunk 4 above)
Comment (braem) [reply mp/src/game/client/momentum/ui/menus/hud_menu_static.cpp]: very true!

File: mp/game/momentum/cfg/menus/mapcmdsmenu_default.vdf [hunk 6]
```diff
@@ -0,0 +1,49 @@
+"Map Commands Menu"
+{
+    "menu_name" "Map Commands Menu"
```
Comment (Gocnak) [inline mp/game/momentum/cfg/menus/mapcmdsmenu_default.vdf:3]: Can remove this now from all the `_default`s.

File: mp/game/momentum/cfg/menus/mapcmdsmenu_default.vdf [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+    "1"
+    {
+        "label"   "#MOM_Menu_MapCommands_Toggle_Wireframes"
```
Comment (Gocnak) [inline mp/game/momentum/cfg/menus/mapcmdsmenu_default.vdf:6]: I didn't see any localization for this?

**Review by Gocnak — CHANGES_REQUESTED** [review]

File: mp/game/momentum/cfg/menus/mapcmdsmenu_default.vdf (hunk 7 above)
Comment (braem) [reply mp/game/momentum/cfg/menus/mapcmdsmenu_default.vdf:6]: Crap, mustve got lost in some rebase somehow.

**Comment by braem** [comment]
Added map cmds localization

image

and removed the redundant "menu_name" field in a new commit as it seems specific.

File: mp/src/game/client/momentum/ui/menus/hud_menu_static_system.cpp [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+{
+    if (args.ArgC() > 1)
+    {
+        g_pHudMenuStaticSystem->ShowMenu(args[1]);
+    }
+    else
+    {
+        Warning("This command expects a parameter that is the hud menu you want to open, by file name");
```
Comment (Gocnak) [inline mp/src/game/client/momentum/ui/menus/hud_menu_static_system.cpp:55]: Missing newline

File: mp/src/game/client/momentum/ui/menus/hud_menu_static_system.cpp [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+void CHudMenuStaticSystem::ReloadMenuList()
+{
+    m_vecMenuNames.RemoveAll();
+
+    FileFindHandle_t findHandle;
+    const char *pFilename = g_pFullFileSystem->FindFirstEx("cfg/menus/*.vdf", "MOD", &findHandle);
+    char pPrevFilename[64];
+    while (pFilename)
+    {
+        if (!FStrEq(pFilename, ".") && !FStrEq(pFilename, ".."))
```
Comment (Gocnak) [inline mp/src/game/client/momentum/ui/menus/hud_menu_static_system.cpp:117]: Does this check need to happen anymore?

**Review by Gocnak — CHANGES_REQUESTED** [review]

File: mp/src/game/client/momentum/ui/menus/hud_menu_static_system.cpp (hunk 9 above)
Comment (braem) [reply mp/src/game/client/momentum/ui/menus/hud_menu_static_system.cpp:117]: Nope! Didn't remove with the wildcard change there.

File: mp/src/game/client/momentum/ui/menus/hud_menu_static_system.cpp (hunk 8 above)
Comment (braem) [reply mp/src/game/client/momentum/ui/menus/hud_menu_static_system.cpp]: I was missing them in a lot of places. Added them now.

**Review by Gocnak — APPROVED** [review]

**Pull request merged**